__init__.py

```diff
@@ -407,8 +412,20 @@ def replace_fn(match):
 
         NT = type("NullTransformer", (Transformer,), methods)
         transformer = {"transformer": NT()}
-
-    parser = Lark(grammar, parser="lalr", start="file", **transformer)
+        
+    if validate_data_only:
+        match = re.search(
```

Comment:
I don't know... as always the risk with these kind of regexes is that they don't follow the grammar, there can be ENDSEC in a string somewhere or comment. Not likely I know, but still can we do this in another way.

Parsing the header is not time consuming and I guess once we have arrived at this point in the VS we have also established that it's valid, can't we do this in another way, like use the grammar to parse the header (again) to find the exact start offset of the data section. And also don't look for the ENDSEC in the file using the grammar, but just see what the parser tells us that the end of the data section is.